From a0d86ed3bc2150ddfad50ab1b2eb09bea7a277d2 Mon Sep 17 00:00:00 2001
From: sin-ack <sin-ack@protonmail.com>
Date: Thu, 24 Apr 2025 20:59:06 +0000
Subject: [PATCH] CMake: Avoid finding liburing

Otherwise, Folly will unconditionally add it to the list of libraries to
link even if io_uring support is otherwise disabled, creating an
implicit dependency on it that can't be tracked by package managers.
---
 CMake/folly-deps.cmake | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 6ce4c679a..b1e20a901 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -139,10 +139,6 @@ find_package(LibAIO)
 list(APPEND FOLLY_LINK_LIBRARIES ${LIBAIO_LIBRARIES})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBAIO_INCLUDE_DIRS})
 
-find_package(LibUring)
-list(APPEND FOLLY_LINK_LIBRARIES ${LIBURING_LIBRARIES})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBURING_INCLUDE_DIRS})
-
 find_package(Libsodium)
 list(APPEND FOLLY_LINK_LIBRARIES ${LIBSODIUM_LIBRARIES})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBSODIUM_INCLUDE_DIRS})
-- 
2.49.0

